packages/orchestrator/internal/sandbox/nbd/path_direct_linux.go

```diff
@@ -19,153 +24,237 @@ import (
 	"github.com/e2b-dev/infra/packages/shared/pkg/telemetry"
 )
 
+const (
+	connections    = 4
+	connectTimeout = 30 * time.Second
+
+	// disconnectTimeout should not be necessary if the disconnect is reliable
+	disconnectTimeout = 30 * time.Second
+)
+
 type DirectPathMount struct {
-	tracer      trace.Tracer
+	tracer   trace.Tracer
+	ctx      context.Context
+	cancelfn context.CancelFunc
+
+	devicePool *DevicePool
+
 	Backend     block.Device
-	ctx         context.Context
-	dispatcher  *Dispatch
-	conn        net.Conn
 	deviceIndex uint32
 	blockSize   uint64
-	cancelfn    context.CancelFunc
-	devicePool  *DevicePool
+
+	dispatchers []*Dispatch
+	socksClient []*os.File
+	socksServer []io.Closer
+
+	handlersWg sync.WaitGroup
 }
 
 func NewDirectPathMount(tracer trace.Tracer, b block.Device, devicePool *DevicePool) *DirectPathMount {
 	ctx, cancelfn := context.WithCancel(context.Background())
 
 	return &DirectPathMount{
-		tracer:     tracer,
-		Backend:    b,
-		ctx:        ctx,
-		cancelfn:   cancelfn,
-		blockSize:  4096,
-		devicePool: devicePool,
+		tracer:      tracer,
+		Backend:     b,
+		ctx:         ctx,
+		cancelfn:    cancelfn,
+		blockSize:   4096,
+		devicePool:  devicePool,
+		socksClient: make([]*os.File, 0),
+		socksServer: make([]io.Closer, 0),
+		deviceIndex: math.MaxUint32,
 	}
 }
 
-func (d *DirectPathMount) Open(ctx context.Context) (uint32, error) {
+func (d *DirectPathMount) Open(ctx context.Context) (retDeviceIndex uint32, err error) {
```

Comment:
why not just like this?
```suggestion
func (d *DirectPathMount) Open(ctx context.Context) (deviceIndex uint32, err error) {
```

Reply:
it would be in conflict with local variable name deviceIndex